fs/copy.ts

```diff
@@ -0,0 +1,256 @@
+  options: CopyOptions
+): Promise<void> {
+  const destStat = await ensureValidCopy(src, dest, options, true);
+
+  if (!destStat) {
+    await ensureDir(dest);
+  }
+
+  if (options.preserveTimestamps) {
+    const srcStatInfo = await Deno.stat(src);
+    await Deno.utime(dest, srcStatInfo.accessed, srcStatInfo.modified);
+  }
+
+  const files = await Deno.readDir(src);
+
+  for (const file of files) {
+    const srcPath = file.path as string;
+    const destPath = path.join(dest, path.basename(srcPath as string));
+    if (file.isDirectory()) {
+      await copyDir(srcPath, destPath, options);
+    } else if (file.isFile()) {
+      await copyFile(srcPath, destPath, options);
+    } else if (file.isSymlink()) {
+      await copySymLink(srcPath, destPath, options);
+    }
+  }
+}
+
+/* copy folder from src to dest synchronously */
+function copyDirSync(src: string, dest: string, options: CopyOptions): void {
+  const destStat: Deno.FileInfo = ensureValidCopySync(src, dest, options, true);
+
+  if (!destStat) {
+    ensureDirSync(dest);
+  }
+
+  if (options.preserveTimestamps) {
+    const srcStatInfo = Deno.statSync(src);
+    Deno.utimeSync(dest, srcStatInfo.accessed, srcStatInfo.modified);
+  }
+
+  const files = Deno.readDirSync(src);
+
+  for (const file of files) {
+    const srcPath = file.path as string;
+    const destPath = path.join(dest, path.basename(srcPath as string));
+    if (file.isDirectory()) {
+      copyDirSync(srcPath, destPath, options);
+    } else if (file.isFile()) {
+      copyFileSync(srcPath, destPath, options);
+    } else if (file.isSymlink()) {
+      copySymlinkSync(srcPath, destPath, options);
+    }
+  }
+}
+
+/**
+ * Copy a file or directory. The directory can have contents. Like `cp -r`.
+ * @param src the file/directory path. Note that if `src` is a directory it will copy everything inside of this directory, not the entire directory itself
+ * @param dest the destination path. Note that if `src` is a file, `dest` cannot be a directory
+ * @param options
+ */
+export async function copy(
+  src: string,
+  dest: string,
+  options: CopyOptions = {}
+): Promise<void> {
+  src = path.resolve(src);
+  dest = path.resolve(dest);
+
+  if (src === dest) {
+    throw new Error("Source and destination must not be the same.");
+  }
+
+  const srcStat = await Deno.lstat(src);
+
+  if (srcStat.isDirectory() && isSubdir(src, dest)) {
+    throw new Error(
+      `Cannot copy '${src}' to a subdirectory of itself, '${dest}'.`
+    );
+  }
+
+  if (srcStat.isDirectory()) {
+    await copyDir(src, dest, options);
+  } else if (srcStat.isFile()) {
+    await copyFile(src, dest, options);
+  } else if (srcStat.isSymlink()) {
+    await copySymLink(src, dest, options);
+  }
+}
+
+/**
+ * Copy a file or directory. The directory can have contents. Like `cp -r`.
+ * @param src the file/directory path. Note that if `src` is a directory it will copy everything inside of this directory, not the entire directory itself
```

Comment:
Wrap at 80 columns (I think prettier should do this)

Reply:
it seems prettier will no wrap comments